### PR TITLE
ARC-788 Fix approximate receive counter

### PR DESCRIPTION
**Pull request by KAbakumovAtl**
* Added `lastAttempt` parameter to the queue execution context
* Fixed receiveCount calcluation
* Added a test

### CODE REVIEW

File: test/unit/sqs/index.test.ts [hunk 1]
```diff
@@ -244,16 +254,56 @@ describe("SqsQueue tests", () => {
 
 			mockErrorHandler.mockImplementation((error: Error, context: Context<TestMessage>) => {
 
-				expect(context.payload.msg).toBe(testPayload.msg);
-				expect(error).toBeInstanceOf(SqsTimeoutError);
+				try {
+					expect(context.payload.msg).toBe(testPayload.msg);
+					expect(error).toBeInstanceOf(SqsTimeoutError);
```
Comment (KAbakumovAtl) [inline test/unit/sqs/index.test.ts:259]: Turns out if expect fails here it doesn't fail the test (cuz exception doesn't reach the top level). So we have to handle it and call fail explicitly

**Review by thombergs — APPROVED** [review]

File: src/sqs/index.ts [hunk 2]
```diff
@@ -318,9 +325,9 @@ export class SqsQueue<MessagePayload> {
 			executionId: uuidv4(),
 			queue: this.queueName})
 
-		const receiveCount = Number(message.MessageAttributes?.ApproximateReceiveCount?.StringValue || "1");
+		const receiveCount = Number(message.Attributes?.ApproximateReceiveCount || "1");
 
-		const context: Context<MessagePayload> = {message, payload, log, receiveCount: receiveCount}
+		const context: Context<MessagePayload> = {message, payload, log, receiveCount: receiveCount, lastAttempt: receiveCount===this.maxAttempts}
```
Comment (bgvozdev) [inline src/sqs/index.ts:330]: nit: Shall we maybe use `>=`, just in case? :)

File: src/sqs/index.ts [hunk 3]
```diff
@@ -351,7 +358,7 @@ export class SqsQueue<MessagePayload> {
 					(errorHandlingResult.skipDlq && context.receiveCount >= this.maxAttempts)) {
 					log.info("Deleting the message hence it reached the maximum amount of retries")
 					await this.deleteMessage(message, log)
-				} else if (errorHandlingResult.retryDelaySec) {
+				} else if (errorHandlingResult.retryDelaySec !== undefined /*zero delay supported*/) {
```
Comment (bgvozdev) [inline src/sqs/index.ts:361]: nit: it took me 15-20 seconds to digest the comment. Might something like `"zero seconds delay is also supported" ` be more clear?

**Review by bgvozdev — APPROVED** [review]

File: src/sqs/index.ts (hunk 2 above)
Comment (KAbakumovAtl) [reply src/sqs/index.ts:330]: Oh yes, exactly! In case of DLQ replay

**Review by bgvozdev — APPROVED** [review]

**Review by thombergs — APPROVED** [review]

**Pull request merged**